### PR TITLE
Rozeta cod Vignere cipher

### PR DESCRIPTION
**Pull request by Bl4ckb0ne**

### CODE REVIEW

File: examples/rosettacode/random_numbers.nit [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env nit
+#
+# This file is part of NIT ( http://www.nitlanguage.org ).
+# This program is public domain
+
+# Task: Random numbers
+#
+# See: <http://rosettacode.org/wiki/Random_numbers>
+
```
Comment (privat) [inline examples/rosettacode/random_numbers.nit]: remove the empty line

**Comment by ventilooo** [comment]
+1

**Comment by privat** [comment]
you should do a distinct branch, so remove `Random numbers from rosettacode`

**Comment by Bl4ckb0ne** [comment]
Done

**Comment by privat** [comment]
please squash the commits into a single one.

**Comment by Bl4ckb0ne** [comment]
Done @privat , et plus de nitcoins!

File: examples/rosettacode/vignere_cipher.nit [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env nit
+#
+# This file is part of NIT ( http://www.nitlanguage.org ).
+# This program is public domain
+
+# Task: Vigenère cipher
+#
+# See: <http://rosettacode.org/wiki/Vigenère_cipher>
+
```
Comment (privat) [inline examples/rosettacode/vignere_cipher.nit]: remove this blang line

File: examples/rosettacode/vignere_cipher.nit [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+module vignere_cipher
+
+fun encrypt(src:String, key:String):String do
```
Comment (privat) [inline examples/rosettacode/vignere_cipher.nit]:
~~~nit
fun encrypt(src, key: String): String do
~~~

File: examples/rosettacode/vignere_cipher.nit [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    var out = new Buffer
+    var j:Int = 0
```
Comment (privat) [inline examples/rosettacode/vignere_cipher.nit]: useless type

File: examples/rosettacode/vignere_cipher.nit [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    for i in [0..src.length-1] do
+        var c:Char = src[i]
```
Comment (privat) [inline examples/rosettacode/vignere_cipher.nit]: useless type

File: examples/rosettacode/vignere_cipher.nit [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            c = c.to_upper
+        else if c < 'A' or c > 'Z' then
+            continue
+        end
+
+        out.add(((c.ascii + key[j].ascii - 2*'A'.ascii) % 26 + 'A'.ascii).ascii)
+        j = (j + 1) % key.length
+    end
+
+    return out.write_to_string
```
Comment (privat) [inline examples/rosettacode/vignere_cipher.nit]: usually we use `to_s` to transform a buffer into a string. in buffer, `write_to_string` is just an alias of `to_s` if a remember correctly

File: examples/rosettacode/vignere_cipher.nit [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+end
+
+fun decrypt(src:String, key:String):String do
```
Comment (privat) [inline examples/rosettacode/vignere_cipher.nit]: same, ...`(src, key: String)`...

File: examples/rosettacode/vignere_cipher.nit [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    end
+
+    return out.write_to_string
+end
+
+# Main part
+var str:String = "All your base are belong to us"
+var key:String = "CRYPTONIT"
+var code:String = encrypt(str, key)
+var decode:String = decrypt(code, key)
```
Comment (privat) [inline examples/rosettacode/vignere_cipher.nit]: useless types

**Comment by Bl4ckb0ne** [comment]
Question for @privat , why are the types useless? Isn't the code more clear with them?

**Comment by R4PaSs** [comment]
It's useless in the sense that it displays redundant information, and whether or not the code is clearer with them is up to a personal point of view (though we advise using the implicit type form).

Since the assignment already gives the type information, you don't really need to add manual type information, it can be inferred by the compiler, and a sane programmer can easily see what the static type of a variable is.

Another problem to consider is that, by restricting the static type of a variable on a special type, you might render valid code at one point invalid should the type of whatever you are assigning your variable to change while the services are still present.

Imagine you get your object through a method call, should the return type of the method change for renaming purposes for instance, your code will become invalid, whilst if you left the typing responibility to the compiler, no issues would have ensued.

File: examples/rosettacode/vignere_cipher.nit (hunk 2 above)
Comment (privat) [inline examples/rosettacode/vignere_cipher.nit]: this is not done yet

**Comment by privat** [comment]
Your spacing and indentation is weird, I suggest that you run [nitpretty](http://nitlanguage.org/tools/nitpretty/) on your program

**Comment by Bl4ckb0ne** [comment]
It might be my vim setup who puts 4 spaces for a tabulation instead of 8.
I'll fix that

**Pull request closed without merging**